apps/dotcom/client/src/tla/app/TldrawApp.ts

```diff
@@ -210,26 +214,42 @@ export class TldrawApp {
 
 		const myFileIds = new Set<string>([...objectMapKeys(myFiles), ...objectMapKeys(myStates)])
 
-		const nextRecentFileOrdering = []
+		const nextRecentFileOrdering: {
+			fileId: TlaFile['id']
+			isPinned: boolean
+			date: number
+		}[] = []
 
 		for (const fileId of myFileIds) {
 			const file = myFiles[fileId]
 			const state = myStates[fileId]
 			if (!file || !state) continue
 			const existing = this.lastRecentFileOrdering?.find((f) => f.fileId === fileId)
-			if (existing) {
+			if (existing && existing.isPinned === state.isPinned) {
```

Comment:
Important: while we want the list to be stable with regard to recent edits (ie we don't want editing a file to change its order in the sidebar until the user reloads), we _do_ want the pinned state to change the appearance of the sidebar immediately.